website_sale_order_type/models/website.py

```diff
@@ -0,0 +1,21 @@
+# Copyright 2023 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+
+
+class Website(models.Model):
+    _inherit = "website"
+    _description = "Allowing sale_order_type to work with website_sale."
+
+    def sale_get_payment_term(self, partner):
+        payment_term_id = super().sale_get_payment_term(partner)
+        if partner.sale_type.payment_term_id:
+            payment_term_id = partner.sale_type.payment_term_id.id
+        return payment_term_id
```

Comment:
And the same for the other:

```suggestion
        if partner.sale_type.payment_term_id:
            return partner.sale_type.payment_term_id.id
        return super().sale_get_payment_term(partner)
```

Reply:
thanks!